common/src/warp_runner/manager/commands/multipass_commands.rs

```diff
@@ -118,9 +121,30 @@ pub async fn handle_multipass_cmd(cmd: MultiPassCmd, warp: &mut super::super::Wa
         MultiPassCmd::CreateIdentity { .. } | MultiPassCmd::TryLogIn { .. } => {
             // do nothing and drop the rsp channel
         }
-        MultiPassCmd::RequestFriend { did, rsp } => {
-            let r = warp.multipass.send_request(&did).await;
-            let _ = rsp.send(r);
+        MultiPassCmd::RequestFriend { id, rsp } => {
+            // First attempt using a did
+            if let Ok(some) = DID::from_str(id.as_str()) {
+                let r = warp.multipass.send_request(&some).await;
+                let _ = rsp.send(r);
+                return;
+            }
+            // Invalid did. Try using it as a username (+ short did)
+            let r = warp.multipass.get_identity(Identifier::Username(id)).await;
```

Comment:
You may want to add something to confirm that there is a short id attached because using `Identifier::Username` will accept with or without but if its with the short id, it would single out to one item in the array rather than multiple if you were to use it without and there is no short id.

Reply:
> i did leave some comments but overall its a good start, but the problem is that if there is a mistake in the placement of the did key, it would automatically attempt to use it as a username, which may not be the correct way to go about it. It might be best to either use a separate command to possibly get the identity and pass the did along to send the request, use something like `Either` from the `either` crate and use some logic before hand to identify if its a possible username or did key (eg it may be a did if it begins with `did:key`, its a specific/fixed length, while username in uplink shouldnt exceed 32 characters, etc).

from what im seeing username can go anywhere from 4-64 chars so the only way would be to check if it starts with `did:key`

Reply:
added something that 
1. checks if the input contains a hash symbol
2. if the second part matches the length of a short id